gammapy/modeling/models/temporal.py

```diff
@@ -662,3 +663,131 @@ def integral(self, t_min, t_max):
+    """Temporal phase curve model.
+
+    A timing solution is used to compute the phase corresponding to time and
+    a template phase curve is used to determine the associated ``norm``.
+
+    The phasecurve is given as a table with columns ``phase`` and ``norm``.
+
+    The ``norm`` is supposed to be a unit-less multiplicative factor in the model,
+    to be multiplied with a spectral model.
+
+    The model does linear interpolation for times between the given ``(phase, norm)`` values.
+
+    The implementation currently uses `scipy.interpolate. InterpolatedUnivariateSpline`,
+    using degree ``k=1`` to get linear interpolation.
+    This class also contains an ``integral`` method, making the computation of
+    mean fluxes for a given time interval a one-liner.
+
+    Parameters
+    ----------
+    table : `~astropy.table.Table`
+        A table with 'PHASE' vs 'NORM'
+    filename : str
+        The name of the file containing the phase curve
+    t_ref : `~astropy.units.Quantity`
+        The reference time in mjd
+    phi_ref : `~astropy.units.Quantity`
+        The phase at reference time
+    f0 : `~astropy.units.Quantity`
+        The frequency at t_ref in s-1
+    f1 : `~astropy.units.Quantity`
+        The frequency derivative at t_ref in s-2
+    f2 : `~astropy.units.Quantity`
+        The frequency second derivative at t_ref in s-3
+    """
+
+    tag = ["TemplatePhaseCurveTemporalModel", "template-phase"]
+    _t_ref_default = Time(48442.5, format="mjd")
+    _phi_ref_default = 0
+    _f0_default = 29.946923 * u.s**-1
+    _f1_default = 0 * u.s**-2
+    _f2_default = 0 * u.s**-3
+
+    t_ref = Parameter("t_ref", _t_ref_default.mjd, unit="day", frozen=True)
+    phi_ref = Parameter("phi_ref", _phi_ref_default, unit="", frozen=True)
+    f0 = Parameter("f0", _f0_default, frozen=True)
+    f1 = Parameter("f1", _f1_default, frozen=True)
+    f2 = Parameter("f2", _f2_default, frozen=True)
+
+    def __init__(self, table, filename=None, **kwargs):
+        self.table = table
+        if filename is not None:
+            filename = str(make_path(filename))
+        self.filename = filename
+        super().__init__(**kwargs)
+
+    @classmethod
+    def read(
+        cls,
+        path,
+        t_ref=_t_ref_default.mjd * u.d,
+        phi_ref=_phi_ref_default,
+        f0=_f0_default,
+        f1=_f1_default,
+        f2=_f2_default,
+    ):
+        """Read phasecurve model table from FITS file.
+
+        Beware : this does **not** read parameters.
+        They will be set to defaults.
+
+        Parameters
+        ----------
+        path : str or `~pathlib.Path`
+            filename with path
+        """
+        filename = str(make_path(path))
+        return cls(
+            Table.read(filename),
+            filename=filename,
+            t_ref=t_ref,
+            phi_ref=phi_ref,
+            f0=f0,
+            f1=f1,
+            f2=f2,
+        )
+
+    def write(self, path=None, overwrite=False):
```

Comment:
The ``write`` method stores only the table, not the other parameters. What is the logic behind? 
In parallel, the ``read`` method ingests the table stored in the filename and the other parameters have to be put in argument. So the whole is coherent. 
But, why can not the full model be stored? When manipulating the dictionary it seems that everything is handled...